resources/js/features/root/layout/Navbar.tsx

```diff
@@ -1,73 +1,130 @@
+
+  const facilityUrl = () => facilitiesQuery.data?.find((facility) => facility.id === facilityId())?.url;
+
+  const sectionItems = createMemo(() => getSectionItems(facilityUrl()));
 
   return (
     <aside class={cx(s.sidebar)}>
-      <FullLogo />
-      <nav class="flex-1">
-        <Show when={params.facilityUrl}>
-          <For each={NAVBAR_ITEMS}>
-            {(item) => (
-              <A
-                href={item.hrefFn({location, params})}
-                class="mb-2 p-4 rounded-lg flex flex-row items-center gap-3 hover:bg-white"
-                activeClass="bg-white"
-              >
-                <item.Icon size="25" />
-                <span>{item.title}</span>
-              </A>
-            )}
-          </For>
+      <div class={cx("flex flex-row justify-between py-4 px-8 bg-inherit")}>
+        <img src="/img/memo_logo.svg" class="h-14" />
+        <img src="/img/cpd_children_logo.svg" class="h-12" />
+      </div>
+      <nav class={cx("flex-1 px-8 py-4 overflow-y-auto", s.navScroll)}>
+        <Show when={facilityUrl()}>
+          <NavigationSection
+            facilityUrl={facilityUrl()}
+            roles={["verified"]}
+            items={sectionItems().looseFacilityItems}
+          />
+        </Show>
+        <NavigationSection items={sectionItems().looseItems} />
+        <NavigationSection roles={["globalAdmin"]} items={sectionItems().system} title="System" />
+        <Show when={facilityUrl()}>
+          <NavigationSection
+            facilityUrl={facilityUrl()}
+            roles={["facilityAdmin"]}
+            items={sectionItems().facility}
+            title="Placówka"
+          />
+          <NavigationSection
+            facilityUrl={facilityUrl()}
+            roles={["facilityStaff"]}
+            items={sectionItems().work}
+            title="Moja praca"
+          />
         </Show>
-        <hr class="my-4" />
-        <A
-          href="/admin"
-          class="mb-2 p-4 rounded-lg flex flex-row items-center gap-3 hover:bg-white"
-          activeClass="bg-white"
-        >
-          <HiSolidUserGroup size="25" />
-          <span>Administracja</span>
-        </A>
-        <A
-          href="/help"
-          class="mb-2 p-4 rounded-lg flex flex-row items-center gap-3 hover:bg-white"
-          activeClass="bg-white"
-        >
-          <HiSolidInformationCircle size="25" />
-          <span>Pomoc</span>
-        </A>
       </nav>
     </aside>
   );
 };
 
-export type HrefFnArgs = {
-  params: {facilityUrl?: string};
-  location: Location;
-};
-
-export type NavbarItem = {
-  Icon: IconTypes;
-  hrefFn: (args: HrefFnArgs) => AnchorProps["href"];
-  title: string;
-};
-
-const NAVBAR_ITEMS: NavbarItem[] = [
-  {
-    Icon: HiSolidUserGroup,
-    hrefFn: ({params: {facilityUrl}}) => {
-      if (facilityUrl) return `/${facilityUrl}/admin`;
-      return "/help";
+const getSectionItems = (
+  facilityUrl?: string,
+): {
+  system: NavigationSectionProps["items"];
```

Comment:
a.k.a. `NavigationItemProps[]`, może jednak? Żebym nie musiał się tam wklikiwać i sprawdzać szczo ce.

Reply:
ja jestem zwolennikiem takiej formy. Jakby typ items się zmienił na inny, to tutaj też musisz zmieniać, a tak to typ jest od razu znany i jedyne błędy mogą się pojawić przy przekazywaniu propsów

Reply:
No tak, ale jeśli ten typ się zmieni to nie byłoby miło gdyby kompilator uświadomił człowiekowi że typ propsów się zmienia? Jak dla mnie to typ propsów raczej nie jest przezroczysty, podobnie jak raczej nie piszesz `function a(x:Parameters[typeof b][0])` tylko dlatego że a przekazuje ten x do b. Co najwyżej wyciągasz i nazywasz ten typ żeby go móc użyć w tych dwóch miejscach.

A teraz ktoś chce użyć tego komponentu (co jest dużo częstszym usecasem niż to że ten typ się zmienia) to musi kliknąć dodatkowo tam żeby sprawdzić co to jest. To jest podobnego typu zaciemnianie co ten Omit jak dla mnie.